csrc/device_lower/analysis/tensor_memory.cpp

```diff
@@ -13,27 +13,82 @@
 
 namespace nvfuser {
 
+// See note [Tensor Memory Allocation] for the overall design.
 TensorMemoryInfo computeTMemInfo(Fusion* fusion) {
-  bool found = false;
+  TensorMemoryInfo result;
+
+  // Step 1: partition the tensors. Each partition of tensor will become a
```

Comment:
Wanted to clarify here. Does this partition mean partitioning of a list of tensors or partitioning of a single tensor? I think it's the former but wanted to confirm.

Reply:
Actually, reading the code below makes it obvious it's the former.

Reply:
Yes, the former